.github/workflows/install_deps/action.yml

```diff
@@ -0,0 +1,113 @@
+name: Install Dependencies
+description: |
+  This workflow installs Python, PennyLane and all its dependencies. If a
+  requirements file is provided, it will upload the results of pip freeze
+  as well.
+inputs:
+  python_version:
+    description: The version of Python to use in order to run unit tests
+    required: false
+    default: '3.9'
+  install_jax:
+    description: Indicate if JAX should be installed or not
+    required: false
+    default: 'true'
+  jax_version:
+    description: The version of JAX to install for any job that requires JAX
+    required: false
+    default: 0.4.23
+  install_tensorflow:
+    description: Indicate if TensorFlow should be installed or not
+    required: false
+    default: 'true'
+  tensorflow_version:
+    description: The version of TensorFlow to install for any job that requires TensorFlow
+    required: false
+    default: 2.15.0
+  install_pytorch:
+    description: Indicate if PyTorch should be installed or not
+    required: false
+    default: 'true'
+  pytorch_version:
+    description: The version of PyTorch to install for any job that requires PyTorch
+    required: false
+    default: 2.2.0
+  install_pennylane_lightning_master:
+    description: Indicate if PennyLane-Lightning should be installed from the master branch
+    required: false
+    default: 'true'
+  additional_pip_packages:
+    description: Additional packages to install. Values will be passed to pip install {value}
+    required: false
+    default: ''
+  requirements_file:
+    description: File name to store stable version of requirements for a test group
+    required: false
+    default: ''
+
+runs:
+  using: composite
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '${{ inputs.python_version }}'
+
+    - name: Upgrade PIP
+      shell: bash
+      run: pip install --upgrade pip && pip install wheel --upgrade
+
+    - name: Install PennyLane dependencies
+      shell: bash
+      run: |
+        pip install -r requirements-ci.txt --upgrade
+        pip install -r requirements-dev.txt --upgrade
+
+    - name: Install PyTorch
+      shell: bash
+      if: inputs.install_pytorch == 'true'
+      env:
+        TORCH_VERSION: ${{ inputs.pytorch_version != '' && format('=={0}', inputs.pytorch_version) || '' }}
+      run: pip install "torch${{ env.TORCH_VERSION }}" -f https://download.pytorch.org/whl/torch_stable.html
```

Comment:
Curious why you're not installing from PyPi.

Reply:
torch is always weird with installation idk. this whole action is just copy-pasted from unit-test.yml